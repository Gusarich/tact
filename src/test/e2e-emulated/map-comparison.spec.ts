import { Dictionary, toNano } from "@ton/core";
import { ContractSystem } from "@tact-lang/emulator";
import { __DANGER_resetNodeId } from "../../grammar/ast";
import { MapComparisonTestContract } from "./contracts/output/map-comparison_MapComparisonTestContract";

describe("map-comparison", () => {
    beforeEach(() => {
        __DANGER_resetNodeId();
    });
    it("should implement map comparison correctly", async () => {
        // Init
        const system = await ContractSystem.create();
        const treasure = system.treasure("treasure");
        const contract = system.open(
            await MapComparisonTestContract.fromInit(),
        );
        const tracker = system.track(contract.address);

        // Test
        {
            const m1: Dictionary<bigint, bigint> = Dictionary.empty(
                Dictionary.Keys.BigInt(256),
                Dictionary.Values.BigInt(256),
            );
            m1.set(1n, 2n);
            m1.set(3n, 4n);
            const m2: Dictionary<bigint, bigint> = Dictionary.empty(
                Dictionary.Keys.BigInt(256),
                Dictionary.Values.BigInt(256),
            );
            m2.set(1n, 2n);
            m2.set(3n, 4n);
            await contract.send(
                treasure,
                { value: toNano("10") },
                {
                    $$type: "CompareIntInt",
                    m1,
                    m2,
                },
            );
            await system.run();
            expect(tracker.collect()).toMatchSnapshot();
        }
    });
});
