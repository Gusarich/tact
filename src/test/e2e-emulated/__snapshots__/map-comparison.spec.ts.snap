// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map-comparison should implement map comparison correctly 1`] = `
[
  {
    "$seq": 0,
    "events": [
      {
        "$type": "deploy",
      },
      {
        "$type": "received",
        "message": {
          "body": {
            "cell": "x{602DB8CDE_}
 x{CFF}
  x{500000000000000000000000000000000000000000000000000000000000000014_}
  x{500000000000000000000000000000000000000000000000000000000000000024_}
 x{CFF}
  x{500000000000000000000000000000000000000000000000000000000000000014_}
  x{500000000000000000000000000000000000000000000000000000000000000024_}",
            "type": "cell",
          },
          "bounce": true,
          "from": "@treasure(treasure)",
          "to": "kQB_fk7kQdo5QXLBsDT5CiTCeaSICOBKsOIkviCSYGKxbfgw",
          "type": "internal",
          "value": "10",
        },
      },
      {
        "$type": "failed",
        "errorCode": 9,
        "errorMessage": "Cell underflow",
      },
      {
        "$type": "sent-bounced",
        "message": {
          "body": {
            "cell": "x{FFFFFFFF602DB8CDE_}",
            "type": "cell",
          },
          "bounce": false,
          "from": "kQB_fk7kQdo5QXLBsDT5CiTCeaSICOBKsOIkviCSYGKxbfgw",
          "to": "@treasure(treasure)",
          "type": "internal",
          "value": "9.996081",
        },
      },
    ],
  },
]
`;
